astropy/io/votable/tests/test_table.py

```diff
@@ -349,6 +349,45 @@ def test_stored_parquet_votable(format):
     assert stored_votable["sfr"].unit == u.solMass / u.year
 
 
+def test_write_jybeam_unit(tmp_path, recwarn):
```

Comment:
Please do review carefully whether the tested behavior makes sense to you.

Might wanna ignore the double run rabbit hole. I tried tweaking 

https://github.com/astropy/astropy/blob/07df8c90853eb454955025f06f908f9d88b70891/astropy/io/votable/exceptions.py#L66

or calling

https://github.com/astropy/astropy/blob/07df8c90853eb454955025f06f908f9d88b70891/astropy/io/votable/table.py#L409

but that just broke even more tests...

Reply:
Update: double run should be handled now via `recwarn`. Please do still review carefully.